subsys/bluetooth/controller/ll_sw/pdu.h

```diff
@@ -809,11 +809,19 @@ struct pdu_data {
 	uint8_t nesn:1;
 	uint8_t sn:1;
 	uint8_t md:1;
+#if defined(CONFIG_BT_CTLR_DF_CONN_CTE_TX)
 	uint8_t cp:1;
 	uint8_t rfu:2;
+#else
+	uint8_t rfu:3;
+#endif
```

Comment:
I didn't do it conditionally compiled to avoid unnecessary `#if defined` in other places.
The split `rfu` field is compliant with former BT Core Spec versions, because then CP bit is always set to zero (direction finding is disabled).
There is not much overhead when use of CP bit separately (or none).

Maybe revert is back?

Reply:
I think compiler will optimize it in case the initialization is done with a constant e.g.
```
p->ll_id = PDU_DATA_LLID_DATA_CONTINUE;
p->cp = 0U;
p->rfu = 0U;
```
should be optimized to single write operation. If there is a dependency on some variable e.g.
```
extern int foo;
p->ll_id = PDU_DATA_LLID_DATA_CONTINUE;
p->cp = foo;
p->rfu = 0U;
```
Then compiler will emit separate instructions to store `foo` value in `p->cp`.

BTW,
I did some checks in *.lst files and it looks that this is the compiler behavior.

Reply:
In the example code, you have kept `cp` and `rfu` being initialized one after the other but that is not the case, `p->rfu = 0U` has to be set for both first and continue PDUs (`cp`, `rfu` and `resv` fields are not initialized in HCI/ULL later, but be initialized in LLL based on CTE enabled, right? ) .

In the fixes in this PR, `cp` and `resv` fields are not modified for first PDU and only reset for continue PDUS. Whereas `rfu` fields is initialized for both first and continue PDUs.

Reply:
> cp, rfu and resv fields are not initialized in HCI/ULL later, but be initialized in LLL based on CTE enabled, right?

In case of CTE these fields are initialized in ULL and interpreted in LLL to configure CTE in radio.

> In the fixes in this PR, cp and resv fields are not modified for first PDU and only reset for continue PDUs. Where as rfu fields is initialized for both first and continue PDUs.

Why `rfu` field has to be initialized in LLL for first and continue PDUs?

Reply:
> In case of CTE these fields are initialized in ULL and interpreted in LLL to configure CTE in radio.

and 

> Why rfu field has to be initialized in LLL for first and continue PDUs?

I prefer to keep these fields be assigned in LLL as PDU header bit fields are mostly vendor specific interpretations

Reply:
I accept it due to: 
> I prefer to keep these fields be assigned in LLL as PDU header bit fields are mostly vendor specific interpretations